gdpr_assist/models.py

```diff
@@ -182,21 +184,27 @@ def get_export_filename(self):
         )
 
 
+class PrivacyAnonymised(models.Model):
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    object_id = models.PositiveIntegerField()
+    anonymised_object = GenericForeignKey('content_type', 'object_id')
+
+
 class PrivacyModel(models.Model):
     """
-    An abstract model base class with support for anonymising data
+        An abstract model base class with support for anonymising data.
     """
-    anonymised = models.BooleanField(default=False)
+    anonymised_relation = GenericRelation(PrivacyAnonymised)
 
     def anonymise(self, force=False):
         # Only anonymise things once to avoid a circular anonymisation
-        if self.anonymised and not force:
+        if self.is_anonymised() and not force:
             return
 
         pre_anonymise.send(sender=self.__class__, instance=self)
 
         # Anonymise data
-        self.anonymised = True
+        PrivacyAnonymised.objects.create(anonymised_object=self)
```

Comment:
Might be worth adding a param to here - bulk i.e. to return to a bulk_create for these objects.

Reply:
Added bulk option.